expenser-server/middleware/auth.go

```
@@ -0,0 +1,64 @@
+			respondWithJson(w, err)
+			return
+		}
+
+		userID, err := auth.ValidateJWT(token, []byte(jwtSecret))
+		if err != nil {
+			respondWithJson(w, err)
+			return
+		}
+
+		// Add userID to request context
+		ctx := context.WithValue(r.Context(), "userID", userID)
```

golangci-lint (1.61.0)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)